admin/install/install.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+// The install binary installs sandboxfs and all of its support files.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bazelbuild/sandboxfs/internal/shell"
+)
+
+const (
+	// Relative path to the runfiles directory for the install script.  Used if the script is
+	// run by hand from the workspace directory, without using "bazel run".
+	runfiles = "bazel-bin/admin/install/install.runfiles/sandboxfs"
+
+	// Path to the built sandboxfs binary relative to the runfiles directory.
+	sandboxfsBin = "cmd/sandboxfs/sandboxfs"
+
+	// Desired permissions for installed files.
+	dataPerm = 0644
+	execPerm = 0755
+	dirPerm  = 0755
+)
+
+// installSpec contains the details for the installation of one source file.
+type installSpec struct {
+	// sourceFile contains the relative path to the source file to be installed.  If ran with
+	// Bazel, all source files must be present in the runfiles directory of this tool, which
+	// means they all must be listed as data dependencies.
+	sourceFile string
+
+	// targetDir indicates the relative path from the prefix where the file will be installed.
+	// The name of the installed file matches the name of the source file.
+	targetDir string
+
+	// mode indicates the desired permissions of the installed file.
+	mode os.FileMode
+}
+
+// installData specifies the files to be installed by this tool.
+var installData = []installSpec{
+	{sandboxfsBin, "bin", execPerm},
+
+	{"AUTHORS", "share/doc/sandboxfs", dataPerm},
+	{"CONTRIBUTING.md", "share/doc/sandboxfs", dataPerm},
+	{"CONTRIBUTORS", "share/doc/sandboxfs", dataPerm},
+	{"INSTALL.md", "share/doc/sandboxfs", dataPerm},
+	{"LICENSE", "share/doc/sandboxfs", dataPerm},
+	{"NEWS.md", "share/doc/sandboxfs", dataPerm},
+	{"README.md", "share/doc/sandboxfs", dataPerm},
+
+	{"cmd/sandboxfs/sandboxfs.1", "share/man/man1", dataPerm},
+}
+
+// doInstall performs the installation of all given data files into the prefix.
+func doInstall(sourceDir string, prefix string, data []installSpec) error {
+	for _, spec := range data {
```

Comment:
installData instead of data?

or filesToInstall ?

Reply:
`installData` might be nice, but I'm wary of shadowing global symbols because that can lead to more confusion. `filesToInstall` sounds nice too, but then I'd also want to rename the global to match this new name so the same issue would arise.

I feel that `data` here is not too bad given that the type definition is immediately before use, the type is descriptive enough, and this is a private function (so the text wouldn't end up in godoc).

Don't feel too strongly about this, so if you have a better suggestion to avoid the shadowing issue...

Reply:
Personally I would rename both to filesToInstall and move the global version of filesToInstall into main

Reply:
Done.